code/concepts/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+
+# Set up the project.
+cmake_minimum_required( VERSION  3.12 )
+project( concepts LANGUAGES CXX )
+
+# Set up the compilation environment.
+include( "${CMAKE_CURRENT_SOURCE_DIR}/../CompilerSettings.cmake" )
+include( "${CMAKE_CURRENT_SOURCE_DIR}/../SolutionTarget.cmake" )
+
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# Create the user's executable.
+add_executable( concepts "concepts.cpp" )
+
+# Create the "solution executable".
+add_executable( concepts.sol1 EXCLUDE_FROM_ALL "solution/concepts.sol1.cpp" )
+add_dependencies( solution1 concepts.sol1 )
+add_executable( concepts.sol2 EXCLUDE_FROM_ALL "solution/concepts.sol2.cpp" )
+add_dependencies( solution2 concepts.sol2 )
+add_dependencies( solution solution1 solution2 )
```

Comment:
The `add_dependencies` is only to make the solution build if you type `make solution`. That would trigger the build of solution1 and solution2.
The following (untested) configuration might work a bit better. I add two new executables, and add the dependency to make them build with `make solution`:
```suggestion
add_executable( solution1 EXCLUDE_FROM_ALL "solution/concepts.sol1.cpp" )
add_executable( solution2 EXCLUDE_FROM_ALL "solution/concepts.sol2.cpp" )
add_dependencies( solution solution1 solution2 )
```

Reply:
> The `add_dependencies` is only to make the solution build if you type `make solution`. That would trigger the build of solution1 and solution2.

That is exactly what I was expecting, but apparently CMAKE is not happy with a new dependeny which rely onother dependencies (and nto executables).